### PR TITLE
Small code clean ups and refactoring

### PR DESCRIPTION
**Pull request by LazaroHurtado**
Fixing some code logic and inconsistencies I've noticed over time, including one bug that has been flying under the radar for a while now!

### CODE REVIEW

File: src/ops/binary_ops.rs [hunk 1]
```diff
@@ -30,7 +30,9 @@ impl Op for BinaryOps {
     }
 
     fn propagate(&self, source: &Value) {
-        let grad = &source.grad().unwrap();
+        let grad = &source
+            .grad()
+            .unwrap_or_else(|| panic!("Cannot backpropagate when gradient is None"));
```
Comment (LazaroHurtado) [inline src/ops/binary_ops.rs:35]: In the future we would have custom error handling but for now panicking should suffice

**Review by Demba3 — APPROVED** [review]
looks good to me

**Review by Demba3 — APPROVED** [review]
looks good to me

**Review by jrego43 — APPROVED** [review]
LGTM.

**Comment by LazaroHurtado** [comment]
@jrego43 Can you accept the invite to become a collaborator for this repo? I need two approvals from collaborators

**Pull request merged**